File: ThirdPartyAdapters/yandex/yandex/src/main/java/com/google/ads/mediation/yandex/banner/YandexAdSizeProvider.java
/*
 * This file is a part of the Yandex Advertising Network
 *
 * Version for Android (C) 2021 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://legal.yandex.com/partner_ch/
 */
package com.google.ads.mediation.yandex.banner;

import androidx.annotation.Nullable;

import com.yandex.mobile.ads.banner.AdSize;

public class YandexAdSizeProvider {

    @Nullable
    public AdSize getAdSizeFromAdMobAdSize(@Nullable final com.google.android.gms.ads.AdSize adMobAdSize) {
        if (adMobAdSize != null) {
            return new AdSize(adMobAdSize.getWidth(), adMobAdSize.getHeight());
        }

        return null;
    }
}

File: ThirdPartyAdapters/yandex/yandex/src/main/java/com/google/ads/mediation/yandex/banner/YandexAdSizeProvider.kt
package com.google.ads.mediation.yandex.banner

import android.content.Context
import com.google.ads.mediation.yandex.base.AdMobServerExtrasParser
import com.yandex.mobile.ads.banner.BannerAdSize
import com.google.android.gms.ads.AdSize as GoogleAdSize

internal class YandexAdSizeProvider {

    fun getAdSize(
            context: Context,
            serverExtrasParser: AdMobServerExtrasParser,
            adMobAdSize: GoogleAdSize?
    ): BannerAdSize? {
        var adSize = serverExtrasParser.parseAdSize(context)

        if (adSize == null) {
            adSize = getAdSizeFromAdMobAdSize(context, adMobAdSize)
        }

        return adSize
    }

    private fun getAdSizeFromAdMobAdSize(context: Context, adMobAdSize: GoogleAdSize?): BannerAdSize? {
        return adMobAdSize?.let {
            BannerAdSize.fixedSize(context, adMobAdSize.width, adMobAdSize.height)
        }
    }
}
